specs/examples/README.md

```diff
@@ -0,0 +1,89 @@
+- [#325 Set up stackblitz or codepen template with Nimble](https://github.com/ni/nimble/issues/325)
+
+## Implementation / Design
+
+### Frameworks to support
+- Vanilla: TypeScript + SCSS
+    - No explicit examples supporting vanilla JavaScript + CSS with local npm builds. Strongly encourage TS + SCSS.
+    - Examples will use [vite](https://vitejs.dev/)
+        - Supports TS + SCSS out of the box
+        - Web Chapter does not have an alternate recommendation yet for Vanilla apps
+        - Vite works on Stackblitz and local npm builds
+            - Parcel [does not work on Stackblitz](https://github.com/parcel-bundler/parcel/issues/7748)
+            - Rollup requires extra config for TS and SCSS
+            - Webpack aligns with the underlying technology of Angular apps but adds lots of complexity to self manage (Angular generally manages for you)
+- CDN: Vanilla JS and CSS
+    - Uses the CDN builds of packages via unpkg.com
+    - Known issue: fonts are not included in the [existing example](https://github.com/ni/nimble/tree/0fe8cc54f7bd444830e3974969519593b532b083/packages/nimble-components#prototyping-in-a-static-webpage)
+        - We should move all the pre-built cdn bundle stuff to a [separate package](https://twitter.com/justinfagnani/status/1521953590931189760?s=20&t=zejlKVWCbEBce8zYFO7Uyg), ie:
+            - `@ni/nimble-bundle`
+                - alternative names:
+                    - `@ni/nimble-components-cdn`
+                    - `@ni/nimbl-components-bundle`
+                    - `@ni/nimble-cdn-bundle`
+                    - `@ni/nimble-cdn`
+        - Alternative:
+            - We could add a build of our CSS + font assets to the existing nimble-components bundle build which will grow the package size and pose the risk of none cdn users accidentally leveraging the contents (as described in the separate package thread linked above).
+- Angular
+- Blazor
+
+### File structure
+  - examples/
+    - `<framework>`/
+        - README.md: Table of contents for the examples
+        - getting-started/
+            - README.md: Should include a Stackblitz link to open the example immediately in addition to other content
+            - package.json
+            - source/
+        - `<other examples>`: TBD
+
+Notes:
+- A subfolder per `<framework>`, i.e.: `vanilla`, `cdn`, `angular`, `blazor`
+- Each `<framework>` will have a `getting-started` example.
+    - Getting started example will be linked from each library package README
+    - Getting started examples assumes brand new developers who may not have web development experience:
+        - guidance on IDEs
+        - step-by-step instructions
+        - links to learn more about general web topics they need to be aware of (CSS custon properties, shadow slots, etc.)
+- Each example participates in the monorepo build
+- Each example follows Web Chapter conventions:
+    - Linting configured
+    - Testing configured
```

Comment:
I was on the fence about including testing in the examples but should we go even further and include storybook (for vanilla and chromatic)?

Reply:
I guess I'm not quite following. Is the suggestion that each example will have tests that accompany it, and you're also asking if each example should have a storybook? I'm not sure what that means. Do you mean should the examples be part of the Chromatic tests (which happen via Storybook presence)?

If my understanding is correct, I would vote "no" on both accounts. While I think the extra coverage in Chromatic tests would be nice, it still feels a bit odd, and I'm not even sure how that would work. Testing the examples just seems like overkill.

Reply:
The examples are each standalone projects. The idea I was going for is that you can download any of the examples, rename it, but it up for code review and it would have everything you would expect for NI best practices for applications (linting, testing, etc).

So it's not that the examples are tested, it's that the examples have the correct test architecture set-up for someone who is getting started.

Reply:
Given that I think the Storybook aspect might be too much. It's somewhat imposing our will beyond a proper boundary. I think it's sufficient for us to have a doc that could provide the guidance to a user on how to integrate Storybook into their own project, if that was easy enough to document.

Reply:
I lean towards not requiring that each example include tests. We could optionally include them for certain examples (particularly getting started or common application patterns) but they might be a distraction that doesn't add much value for more trivial examples.

Reply:
Yea it does seem unnecessary for all examples. I'll update to say just for the getting started example and just unit tests (not more opionated / env specific like integration / chromatic)